PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,346 @@
+    @objc(RCNormal) case normal = 0
+    /// If the entitlement is under a introductory price period.
+    @objc(RCIntro) case intro = 1
+    /// If the entitlement is under a trial period.
+    @objc(RCTrial) case trial = 2
+}
+
+@objcMembers @objc(RCEntitlementInfo) public class EntitlementInfo: NSObject {
+    /**
+     The entitlement identifier configured in the RevenueCat dashboard
+     */
+    public let identifier: String
+
+    /**
+     True if the user has access to this entitlement
+     */
+    public let isActive: Bool
+
+    /**
+     True if the underlying subscription is set to renew at the end of
+     the billing period (expirationDate). Will always be True if entitlement
+     is for lifetime access.
+     */
+    public let willRenew: Bool
+
+    /**
+     The last period type this entitlement was in
+     Either: RCNormal, RCIntro, RCTrial
+     */
+    public let periodType: PeriodType
+
+    /**
+     The latest purchase or renewal date for the entitlement.
+     */
+    public let latestPurchaseDate: Date? // TODO: This used to be NON-NULL
+
+    /**
+     The first date this entitlement was purchased
+     */
+    public let originalPurchaseDate: Date? // TODO: This used to be NON-NULL
+
+    /**
+     The expiration date for the entitlement, can be `nil` for lifetime access.
+     If the `periodType` is `trial`, this is the trial expiration date.
+     */
+    public private(set) var expirationDate: Date?
+
+    /**
+     The store where this entitlement was unlocked from
+     Either: RCAppStore, RCMacAppStore, RCPlayStore, RCStripe, RCPromotional, RCUnknownStore
+     */
+    public let store: Store
+
+    /**
+     The product identifier that unlocked this entitlement
+     */
+    public let productIdentifier: String
+
+    /**
+     False if this entitlement is unlocked via a production purchase
+     */
+    public let isSandbox: Bool
+
+    /**
+     The date an unsubscribe was detected. Can be `nil`.
+
+     Note: Entitlement may still be active even if user has unsubscribed. Check the `isActive` property.
+     */
+    public private(set) var unsubscribeDetectedAt: Date?
+
+    /**
+     The date a billing issue was detected. Can be `nil` if there is no
+     billing issue or an issue has been resolved.
+
+     Note: Entitlement may still be active even if there is a billing issue.
+     Check the `isActive` property.
+     */
+    public private(set) var billingIssueDetectedAt: Date?
+
+    /**
+     Use this property to determine whether a purchase was made by the current user
+     or shared to them by a family member. This can be useful for onboarding users who have had
+     an entitlement shared with them, but might not be entirely aware of the benefits they now have.
+     */
+    public let ownershipType: PurchaseOwnershipType
+
+    // TODO(post-migration): Make this internal
+    // TODO(cleanup): Codable
+    public init(entitlementId: String,
+                entitlementData: [String: Any],
```

Comment:
this could be constrained at the very least to `AnyObject`, right? Since although the header didn't specify anything, we know it was only used by objective-c

Reply:
and from our usage, we should be able to assume that this is actually always `String` I believe

Reply:
well, the old `parseDate` method does introduce some questions I guess, since it does check to see if the object is a string